### PR TITLE
Combobox: Add onMenuDismiss prop

### PR DESCRIPTION
**Pull request by manueldosal**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Adding an onMenuDismissed prop to ComboBoxes.

#### Focus areas to test

Opening and Closing ComboBoxes

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -240,6 +241,10 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     if (isOpen && !prevState.isOpen && onMenuOpen) {
       onMenuOpen();
     }
+
+    if (!isOpen && prevState.isOpen && onMenuDismissed) {
+      onMenuDismissed();
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: I'll defer to @jspurlin, but shouldn't we put this the _onDismiss method?

File: common/changes/office-ui-fabric-react/Implement_OnDismissMenu_2017-12-12-00-47.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Adding onMenuDismissed for ComboBox",
```
Comment (christiango) [inline common/changes/office-ui-fabric-react/Implement_OnDismissMenu_2017-12-12-00-47.json]: The format of these messages and usually the PR title : is [ComponentName]: description

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (manueldosal) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: It doesn't get called every time we dismiss a menu, for example, when pressing tab

Comment (manueldosal) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: that's the reason I put it here similar to onMenuOpen

Comment (christiango) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Nit: can we make this onMenuDismiss to make it consistent with onMenuOpen?

Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: The name suggested here aligns with the onMenuDismissed for the ContextualMenu.

It does seems weird that this gets called on the next Render and not when we know we are dismissing the menu (e.g. setting the opened state to false). Consider creating a function to set the open state and call onMenuDismissed when you are setting open to false

Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: But the same thing could be said about the onMenuOpen function so this isn't a must, just a consideration... You could actually pull onMenuOpen into the same function if you did that work

Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Talked with @christiango (who added onMenuOpen) offline about this let's keep this code as is for now

**Pull request merged**